js/backup_restore.js

```diff
@@ -636,7 +636,7 @@ function configuration_restore(callback) {
     function configuration_upload(configuration, callback) {
         function upload() {
             var activeProfile = null,
-                profilesN = 3;
```

Comment:
Agreed. The change you suggest should be part of another PR.
This one fixes a crash, I think it should be merged for the time being.